### PR TITLE
Add Sphinx Documentation

### PR DESCRIPTION
**Pull request by ericpien**
**Changes**
- `yfinance/doc` contains `/source` with the configuration and the template rsts.
- Updated README and docstrings

**Current Workflow**
- push to `dev-documented` branch will trigger `yfinance/.github/workflows/deploy_doc.yml` 
- `deploy_doc.yml` configured to build the documentation from `yfinance/doc` and publish the htmls to `documentation` branch's `docs` directory.
- This will update the [documentation](https://ranaroussi.github.io/yfinance/) as set up in the repo's settings/pages.

Relates to https://github.com/ranaroussi/yfinance/discussions/1330

### CODE REVIEW

File: yfinance/multi.py [hunk 1]
```diff
@@ -35,59 +36,44 @@
 
 @utils.log_indent_decorator
 def download(tickers, start=None, end=None, actions=False, threads=True,
-             ignore_tz=None, group_by='column', auto_adjust=False, back_adjust=False,
+             ignore_tz=None, group_by='coluemn', auto_adjust=False, back_adjust=False,
```
Comment (ValueRaider) [inline yfinance/multi.py:39]: typo

File: requirements.txt [hunk 2]
```diff
@@ -9,3 +9,9 @@ frozendict>=2.3.4
 beautifulsoup4>=4.11.1
 html5lib>=1.1
 peewee>=3.16.2
+
+# Sphinx Documentation
```
Comment (ValueRaider) [inline requirements.txt:13]: Not essential for a user install so shouldn't be in this file. At minimum, comment-out.

**Review by ValueRaider — CHANGES_REQUESTED** [review]

**Review by ValueRaider — APPROVED** [review]

**Comment by ValueRaider** [comment]
The action claims success:
https://github.com/ranaroussi/yfinance/actions/runs/11690739308
But link is 404

What is your source set to? yfinance might be the opposite. 
github-page-source_000

**Comment by ericpien** [comment]
My setting is the same and set to Deploy from a branch.

I think I found the problem: Apparently the publishing branch [has to be created by GitHub Actions](https://github.com/peaceiris/actions-gh-pages/issues/647#issuecomment-907667908), not manually created.

Tested and confirmed as below:
1. I manually created a branch `documentation_manual` and [pushed](https://github.com/ericpien/yfinance/commit/1114ed15184ea146fd4a22086ec1251088a76dd4) to my fork's dev-documentation and published to [`documentation_manual/docs`](https://github.com/ericpien/yfinance/tree/documentation_manual/docs). This resulted in the same behavior in which only non-html subset of filers were copied.
2. I did not manually create the branch `documentation_generated` and [pushed](https://github.com/ericpien/yfinance/commit/158f90a5e9445bdb807f65ee224cc33d07a16d82) again to my fork's dev-documentation and published to [`documentation_generated/docs`](https://github.com/ericpien/yfinance/tree/documentation_generated/docs). This correctly copied the full repo including the htmls.

Can I submit another PR to this branch and try to publish to a non-existing branch via GitHub Actions?

**Comment by ValueRaider** [comment]
Understood. I've reverted `documentation` and renamed to `docs_markdown`, so name `documentation` should be free.

**Comment by ericpien** [comment]
Thank you. Are you able to re-run https://github.com/ranaroussi/yfinance/actions/runs/11690726258? If not, I can submit a trivial PR targetting `dev-documented` to retrigger and try to publish to `documentation`.

**Comment by ValueRaider** [comment]
I triggered your action again
https://github.com/ranaroussi/yfinance/actions

But this didn't then trigger a new deployment
https://github.com/ranaroussi/yfinance/deployments/github-pages

**Comment by ericpien** [comment]
I tried to replicate it on my end and noticed **renaming a branch 1. unpublishes the page and 2. reassigns the deploying branch to the renamed branch** so now we need to push to `docs_markdown` to deploy.

So the workflow requires us to publish to a new branch created by GitHub Actions. But we are unable to assign the deploying branch to this new branch.. Are we absolutely restricted from changing the deploying branch/dir?

image

**Comment by ValueRaider** [comment]
Only @ranaroussi can access settings page.

Is an option just dumping the html files into `docs` folder of deploy branch? Like I did here: https://github.com/ValueRaider/yfinance-cache/blob/5ab1bb4b17df9d134e745fb3fa27123b0f9951e2/make-docs.sh#L33

**Comment by ericpien** [comment]
Yeah that works. I worked out an example in my fork: Upon push to `docs_markdown` branch (i.e. [merge PR](https://github.com/ericpien/yfinance/pull/5/)), the htmls are [built](https://github.com/ericpien/yfinance/actions/runs/11696679464) and committed to `docs_markdown` branch's `/docs` [directory](https://github.com/ericpien/yfinance/tree/docs_markdown/docs) then the page is [deployed](https://github.com/ericpien/yfinance/actions/runs/11696696959).

So based on that, I expect the PR https://github.com/ranaroussi/yfinance/pull/2115/ to do the same for `ranaroussi::docs_markdown/docs`

I think this is a pretty good alternative since we can push changes to `docs_markdown` during development, and eventually simply push the new changes from `main`.

**Comment by ericpien** [comment]
For continuity, I'll leave the comment here. 

Following the [PR #2115 to `docs_markdown`](https://github.com/ranaroussi/yfinance/pull/2115), every file needed was built and uploaded to [`docs/`](https://github.com/ranaroussi/yfinance/tree/docs_markdown/docs) but the page build was not triggered in [actions](https://github.com/ranaroussi/yfinance/actions/workflows/pages/pages-build-deployment).

This is odd as a new commit to the renamed publishing branch (i.e. `docs_markdown`) should have triggered a new build. I tested this behavior by renaming my fork's publishing branch to `docs_markdown_renamed` and merging a [PR](https://github.com/ericpien/yfinance/pull/7) that resulted in automatic [rebuild](https://github.com/ericpien/yfinance/actions/runs/11708973356) and documentation is [available](https://ericpien.github.io/yfinance/).

@ranaroussi are you able to give us a peek at the settings/pages configuration like below? 

image

**Comment by ValueRaider** [comment]
While we wait for Ran, for a plan B are you OK with README linking to your fork? Maybe you just need to click `sync fork` to trigger redeploys.

**Comment by ranaroussi** [comment]
Done.

Thank you all for the great

**Comment by ValueRaider** [comment]
@ranaroussi Can you post a screenshot of the current settings in `Build and deployment`?

**Pull request merged**